Try to run an example on windows 11 fails

Try to run an example on Windows with Powershell and I saw the below error:
`..\..\gst\control_source.go:44:162: cannot use _Ctype_ulong(time) (value of type _Ctype_ulong) as _Ctype_ulonglong value in variable declaration`

Just adding that this is also an issue on macos

```
../../gst/control_source.go:44:162: cannot use _Ctype_ulong(time) (value of type _Ctype_ulong) as _Ctype_ulonglong value in variable declaration
```

I've also got two other fails on macos:

```
../../gst/gst_object.go:40:53: cannot use _cgo0 (variable of type unsafe.Pointer) as _Ctype_gpointer value in argument to _Cfunc_g_free
```

and 

```
../../gst/constants.go:42:31: cannot use (_Ciconst_GST_CLOCK_TIME_NONE) (untyped int constant -1) as ClockTime value in variable declaration (overflows)
```

--- a/gst/constants.go
+++ b/gst/constants.go
@@ -39,7 +39,7 @@ const (
 type ClockTimeDiff int64
 
 // ClockTimeNone means infinite timeout or an empty value
-var ClockTimeNone ClockTime = C.GST_CLOCK_TIME_NONE
+var ClockTimeNone ClockTime = 0xffffffffffffffff // Ideally this would be set to C.GST_CLOCK_TIME_NONE but this causes issues on MacOS and Windows
 
 // BufferOffsetNone is a var for no-offset return results.
 var BufferOffsetNone time.Duration = time.Duration(-1)

--- a/gst/control_source.go
+++ b/gst/control_source.go
@@ -41,5 +41,5 @@ func (cs *InterpolationControlSource) SetInterpolationMode(mode InterpolationMod
 }
 
 func (cs *InterpolationControlSource) SetTimedValue(time ClockTime, value float64) bool {
-	return gobool(C.gst_timed_value_control_source_set(C.toGstTimedValueControlSource(cs.Unsafe()), C.ulong(time), C.double(value)))
+	return gobool(C.gst_timed_value_control_source_set(C.toGstTimedValueControlSource(cs.Unsafe()), C.GstClockTime(time), C.double(value)))
 }

--- a/gst/gst_object.go
+++ b/gst/gst_object.go
@@ -37,7 +37,7 @@ func (o *Object) GObject() *glib.Object { return o.InitiallyUnowned.Object }
 func (o *Object) GetName() string {
 	cName := C.gst_object_get_name((*C.GstObject)(o.Instance()))
 	//cName could be NULL which needs to be freed using
-	defer C.g_free(unsafe.Pointer(cName))
+	defer C.g_free((C.gpointer)(unsafe.Pointer(cName)))
 	return C.GoString(cName)
 }
 

